api/v1alpha3/awscluster_types.go

```diff
@@ -148,6 +152,23 @@ type AWSClusterStatus struct {
 	Conditions     clusterv1.Conditions     `json:"conditions,omitempty"`
 }
 
+type S3Bucket struct {
+	// Enabled controls if S3 Bucket should be created as part of cluster provisioning.
+	Enabled bool `json:"enabled,omitempty"`
+
+	// ControlPlaneIAMInstanceProfile is a name of the IAMInstanceProfile, which will be allowed
+	// to read control-plane node bootstrap data from S3 Bucket.
+	ControlPlaneIAMInstanceProfile string `json:"controlPlaneIAMInstanceProfile,omitempty"`
```

Comment:
These should also have `//+optional` IMO

Reply:
`iamInstanceProfile` on `AWSMachineSpec` is not optional, so I just follow it, as it's the same value.

Reply:
Marking as resolved.